homeassistant/components/rfxtrx/__init__.py

```diff
@@ -280,7 +280,9 @@ def _updated_device(event: Event[EventDeviceRegistryUpdatedData]) -> None:
     hass.data[DOMAIN][DATA_RFXOBJECT] = rfx_object
 
     entry.async_on_unload(
-        hass.bus.async_listen(dr.EVENT_DEVICE_REGISTRY_UPDATED, _updated_device)
+        hass.bus.async_listen(
+            dr.EVENT_DEVICE_REGISTRY_UPDATED, _updated_device, run_immediately=False
+        )
```

Comment:
This one has a race in the test where the device goes away too soon because its removed from the registry.

The test is likely bad, but out of scope to fix it here